connectd/connectd.c

```diff
@@ -967,6 +976,23 @@ static void finalize_announcable(struct wireaddr **announcable)
 	}
 }
 
+
+static const struct wireaddr *
+find_local_address(const struct wireaddr_internal *bindings)
+{
+	for (size_t i = 0; i < tal_count(bindings); i++) {
+		if (bindings[i].itype != ADDR_INTERNAL_WIREADDR)
+			continue;
+		if (bindings[i].u.wireaddr.type != ADDR_TYPE_IPV4
+		    && bindings[i].u.wireaddr.type != ADDR_TYPE_IPV6)
+			continue;
+		return &bindings[i].u.wireaddr;
+	}
+	status_failed(STATUS_FAIL_INTERNAL_ERROR,
+		      "No local address found to tell Tor to connect to");
+}
```

Comment:
Make this non static and delete the one in tor_autoservice.c?  Having two is just going to confuse me!

Reply:
Now non static in tor_autoservice.c ( fewer changes )